tests/behaviour/expectations/behaviour.ts

```diff
@@ -132,6 +132,58 @@ export const expectations = flatten(
             Expect.Simple('explicit', ['240'], ['240', '240', '0']),
           ]),
         ]),
+        new Dir('cross_contract_calls', [
+          File.Simple('simple', [Expect.Simple('f', [], ['69', '0'])], 'A'),
+          File.Simple(
+            'simple',
+            [
+              Expect.Simple(
+                'f',
+                ['address@tests/behaviour/contracts/cross_contract_calls/simple.A'],
+                ['69', '0'],
+              ),
+            ],
+            'WARP',
+          ),
+          File.Simple('public_vars', [Expect.Simple('f', [], ['696', '0'])], 'A'),
+          // File.Simple(
```

Comment:
Are these tests meant to be commented out? Also is there a reason that multiple tests on the same file are split into separate uses of `File.Simple`?

Reply:
For the commented tests, a weird error is coming, similar to: ` \' message\': \' Contract address 0x0 is out of range\'})'})` when it executes the line `B.setA_f95a0ad2(__warp_usrid0_a, __warp_usrid2_addrA)` in the Cairo function:
```Cairo
func setB_57be73c5_internal{
      syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt}(
      __warp_usrid2_addrA : felt, __warp_usrid3_addrB : felt) -> ():
  alloc_locals
  WS_WRITE0(__warp_usrid0_a, __warp_usrid3_addrB)
  B.setA_f95a0ad2(__warp_usrid0_a, __warp_usrid2_addrA)
  return ()
end
```
I think the value of `__warp_usrid0_a` is still 0 after the write operation and has to be read using `WS0_READ_felt(__warp_usrid0_a)` operation. Not sure though. 

The reason for multiple uses of `File.Simple` is to deploy each contract from the solc file. Now, `File.Simple` will accept a new optional contract name argument.

Reply:
Ahh I see now. I think this is the same issue Carmen was running into with using state variables in tests. It's fixed in my branch, but in the meantime a workaround should be to read the storage variable into a temporary before passing it to a function. So `a.setA(addrA)` becomes `B temp = a; temp.setA(addrA);` and similarly for other instances where storage variables end up being used like that

Reply:
Done, tests are passing